### PR TITLE
feat: Add item factory

### PR DESCRIPTION
**Pull request by LautaroPetaccio**
This PR adds the item factory. This PR must be merged after #13 is merged.

### CODE REVIEW

File: src/item/ItemFactory.ts [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+   * @param contents - The contents to be filtered taking into consideration the specified body shape.
+   */
+  private removeContentsOfBodyShape(
+    bodyShape: BodyShapeType,
+    contents: Record<string, any>
+  ): Record<string, any> {
+    return Object.keys(contents)
+      .filter(
+        (key) =>
+          !(
```
Comment (nicosantangelo) [inline src/item/ItemFactory.ts:108]: this is a bit hard to spot, maybe we can apply Boole here?

File: src/item/ItemFactory.ts [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+  BodyShapeType,
+  BuiltItem,
+  ItemType,
+  LocalItem,
+  ModelMetrics,
+  WearableBodyShape,
+  WearableCategory
+} from './types'
+
+export class ItemFactory<X extends Content> {
```
Comment (nicosantangelo) [inline src/item/ItemFactory.ts:19]: This is a bit personal, but what do you think about moving private methods after the public ones? I feel like you shouldn't want to see those but it's cumbersome to have to scroll thru a class trying to find the methods that you actually care about

**Review by nicosantangelo — COMMENTED** [review]
Looks awesome

File: README.md [hunk 3]
```diff
@@ -35,3 +35,107 @@ To use the `BuilderClient`, just instantiate the class with the builder-server u
...
+      }
+  })
+```
+
+or deleted from the item using the `withoutRepresentation` method:
+
+```typescript
+const itemFactory = new ItemFactory(oldItem)
+itemFactory.withoutRepresentation(BodyShapeType.FEMALE)
+  })
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply README.md]: Removed it!

File: README.md [hunk 4]
```diff
@@ -35,3 +35,107 @@ To use the `BuilderClient`, just instantiate the class with the builder-server u
...
+```
+
+The metrics must used in the `withRepresentation` method must be computed by the library's user.
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply README.md]: Ups, I forgot to remove the first must. Removed it!

File: src/client/BuilderClient.ts [hunk 5]
```diff
@@ -9,7 +10,11 @@ export class BuilderClient {
   private axios: AxiosInstance
   private readonly AUTH_CHAIN_HEADER_PREFIX = 'x-identity-auth-chain-'
 
-  constructor(url: string, private identity: AuthIdentity) {
+  constructor(
+    url: string,
+    private identity: AuthIdentity,
+    private address: string
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply src/client/BuilderClient.ts:16]: Updated it!

File: src/item/ItemFactory.ts [hunk 6]
```diff
@@ -0,0 +1,450 @@
...
+  private item: LocalItem | null = null
+  private newContent: RawContent<X> = {}
+  private readonly NOT_INITIALIZED_ERROR = 'Item has not been initialized'
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply src/item/ItemFactory.ts]: I wasn't sure about adding a new error just to add a specific message to it, but it is repeated heavily in the code, so I'll add it.

**Review by nicosantangelo — APPROVED** [review]

**Pull request merged**